src/binding.ts

```diff
@@ -0,0 +1,64 @@
+/* --------------------------------------------------------------------------------------------
+ * SonarLint for VisualStudio Code
+ * Copyright (C) 2017-2022 SonarSource SA
+ * sonarlint@sonarsource.com
+ * Licensed under the LGPLv3 License. See LICENSE.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+'use strict';
+
+import * as VSCode from 'vscode';
+
+const SONARLINT_CATEGORY = 'sonarlint';
+const PROJECT_BINDING_PROPERTY = 'connectedMode.project';
+
+export class BindingService {
+
+  private static _instance: BindingService;
+
+  static init(): void {
+    BindingService._instance = new BindingService();
+  }
+
+  static get instance(): BindingService {
+    return BindingService._instance;
+  }
+
+  async updateBinding(workspaceFolderUri: VSCode.Uri, projectBinding: ProjectBinding): Promise<void> {
+    const config = VSCode.workspace.getConfiguration(SONARLINT_CATEGORY, workspaceFolderUri);
+    return config.update(PROJECT_BINDING_PROPERTY, projectBinding, VSCode.ConfigurationTarget.WorkspaceFolder);
+  }
+
+  async deleteBinding(workspaceFolderUri: VSCode.Uri): Promise<void> {
+    const config = VSCode.workspace.getConfiguration(SONARLINT_CATEGORY, workspaceFolderUri);
+    return config.update(PROJECT_BINDING_PROPERTY, undefined, VSCode.ConfigurationTarget.WorkspaceFolder);
+  }
+
+  async getAllBindings(): Promise<Map<string, Map<string, ProjectBinding[]>>> {
+    const bindings = new Map<string, Map<string, ProjectBinding[]>>();
+    for (const folder of VSCode.workspace.workspaceFolders || []) {
+      const config = VSCode.workspace.getConfiguration(SONARLINT_CATEGORY, folder.uri);
+      const binding = config.get<ProjectBinding>(PROJECT_BINDING_PROPERTY);
+      const projectKey = binding.projectKey;
```

Comment:
Here I'm checking against projectKey not for the binding itself because there is a possibility that user will have this in the settings:
```
    "sonarlint.connectedMode.project": {
    
    }
```
And it will be an existing binding but unlikely we want to display such things on frontend.

Reply:
This is not a valid binding: a project key on the server cannot be null. So indeed, I believe it should not be shown in the UI.